### DISCUSSION TITLE
Run time built menu

### DISCUSSION THREAD

**Discussion by 2Skye** [ant-design-blazor/ant-design-blazor#3259] [Q&A]
I'm implementing a run time menu with stringbuilder.append with html tag
![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/123640125/f13e7073-7c76-45c8-bd1f-83a9198ccb9e)

The result of the output string is:
<Menu DefaultSelectedKeys=@(new[] { "1" }) Mode=MenuMode.Inline Theme=MenuTheme.Light>
<MenuItem Key="Supporto">Supporto</MenuItem>
<MenuItem Key="Anagrafiche">Anagrafiche</MenuItem>
<MenuItem Key="Articoli">Articoli</MenuItem>
</Menu>

I use it with @((MarkupString)variable)
but the result is
![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/123640125/924de5c3-992b-4502-9f47-2d498df4353b)

If I copy the string value insert it in the razor page and go to manual field the menu is displayed correct.

I don't understand the two different results.
Does anyone have ideas or has this problem before?

Waiting for your help, I send you my best regards

**Comment by ElderJames** [comment]
Hello @2Skye , the `MarkupString`  is just only for rendering raw HTML content, not for rendering components. See https://github.com/dotnet/aspnetcore/issues/15647

**Comment by 2Skye** [comment]
HI ElderJames

Thanks for the reply.
I saw that in the discussion you mentioned, which dates back to 2018, there was talk of also implementing component rendering.
Hopefully it can be done in the future as creating menus or components at run time is certainly useful.

Good work

**Comment by ElderJames** [comment]
Thank you @2Skye , you can look at the source code how do we implement dynamic menu: 
https://github.com/ant-design-blazor/ant-design-blazor/blob/master/site/AntDesign.Docs/Shared/NavMenu.razor
